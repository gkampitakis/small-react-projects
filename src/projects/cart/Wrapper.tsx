import React, { ReactElement } from 'react';
import { useBodyStyles, useDocTitle } from '../../hooks';
import { AppProvider } from './context';
import App from './App';
import './index.scss';

export default function Wrapper(): ReactElement {
  useDocTitle('Cart');
  useBodyStyles({
    color: '#102a42',
    background: '#f1f5f8'
  });

  return (
    <AppProvider>
      <App />
    </AppProvider>
  );
}
