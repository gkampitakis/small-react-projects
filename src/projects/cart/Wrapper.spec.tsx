import React from 'react';
import { FetchMock } from 'jest-fetch-mock/types';
import { screen, render, waitForElementToBeRemoved } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import Wrapper from './Wrapper';


const fetchMock = (fetch as FetchMock);

describe('Cart', () => {

  beforeEach(fetchMock.disableMocks);

  it('Should render correctly loading', () => {
    const { baseElement } = render(<Wrapper />);

    expect(baseElement).toMatchSnapshot();
  });

  it('Should render correctly after loaded', async () => {
    const { baseElement } = render(<Wrapper />);

    await waitForElementToBeRemoved(screen.getByText(/Loading/i));

    expect(baseElement).toMatchSnapshot();
  });

  it('Should clear cart', async () => {
    const { baseElement } = render(<Wrapper />);

    await waitForElementToBeRemoved(screen.getByText(/Loading/i));

    userEvent.click(screen.getByText(/clear cart/i));

    expect(baseElement).toMatchSnapshot();
  });

  it('Should change amount and cart total', async () => {
    const { baseElement } = render(<Wrapper />);

    await waitForElementToBeRemoved(screen.getByText(/Loading/i));

    const increaseBtn = screen.getAllByRole('button', { name: /increase/i });
    const decreaseBtn = screen.getAllByRole('button', { name: /decrease/i });
    const removeBtn = screen.getAllByRole('button', { name: /remove/i });

    userEvent.click(increaseBtn[0]);
    userEvent.click(decreaseBtn[1]);
    userEvent.click(removeBtn[2]);
    expect(baseElement).toMatchSnapshot();
  });
});
