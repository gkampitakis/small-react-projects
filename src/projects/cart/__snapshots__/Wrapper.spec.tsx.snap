// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Should change amount and cart total 1`] = `
<body
  style="color: rgb(16, 42, 66); background: rgb(241, 245, 248);"
>
  <div>
    <main
      class="cart_project"
    >
      <nav>
        <div
          class="container"
        >
          <h3>
            Cart Project
          </h3>
          <div
            class="amount-container"
          >
            <svg
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 6v2h2l2 12H0L2 8h2V6a6 6 0 1 1 12 0zm-2 0a4 4 0 1 0-8 0v2h8V6zM4 10v2h2v-2H4zm10 0v2h2v-2h-2z"
              />
            </svg>
            <div
              class="amount"
            >
              <p>
                3
              </p>
            </div>
          </div>
        </div>
      </nav>
      <section
        class="cart-container"
      >
        <header>
          <h2>
            Your Bag
          </h2>
        </header>
        <div>
          <article
            class="cart-item"
          >
            <img
              alt="mockTitle0"
              src="mockImg0"
            />
            <div>
              <h4>
                mockTitle0
              </h4>
              <h4
                class="item-price"
              >
                100
              </h4>
              <button
                aria-label="remove"
                class="remove-btn"
              >
                remove
              </button>
            </div>
            <div>
              <button
                aria-label="increase"
                class="amount-btn"
              >
                <svg
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.707 7.05L10 6.343 4.343 12l1.414 1.414L10 9.172l4.243 4.242L15.657 12z"
                  />
                </svg>
              </button>
              <p
                class="amount"
              >
                3
              </p>
              <button
                aria-label="decrease"
                class="amount-btn"
              >
                <svg
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
                  />
                </svg>
              </button>
            </div>
          </article>
        </div>
        <footer>
          <hr />
          <div
            class="cart-total"
          >
            <h4>
              total 
              <span>
                $
                300.00
              </span>
            </h4>
          </div>
          <button>
            clear cart
          </button>
        </footer>
      </section>
    </main>
  </div>
</body>
`;

exports[`Cart Should clear cart 1`] = `
<body
  style="color: rgb(16, 42, 66); background: rgb(241, 245, 248);"
>
  <div>
    <main
      class="cart_project"
    >
      <nav>
        <div
          class="container"
        >
          <h3>
            Cart Project
          </h3>
          <div
            class="amount-container"
          >
            <svg
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 6v2h2l2 12H0L2 8h2V6a6 6 0 1 1 12 0zm-2 0a4 4 0 1 0-8 0v2h8V6zM4 10v2h2v-2H4zm10 0v2h2v-2h-2z"
              />
            </svg>
            <div
              class="amount"
            >
              <p>
                0
              </p>
            </div>
          </div>
        </div>
      </nav>
      <section
        class="cart-container"
      >
        <header>
          <h2>
            Your Bag
          </h2>
          <h4
            class="empty-cart"
          >
            is currently empty
          </h4>
        </header>
      </section>
    </main>
  </div>
</body>
`;

exports[`Cart Should render correctly after loaded 1`] = `
<body
  style="color: rgb(16, 42, 66); background: rgb(241, 245, 248);"
>
  <div>
    <main
      class="cart_project"
    >
      <nav>
        <div
          class="container"
        >
          <h3>
            Cart Project
          </h3>
          <div
            class="amount-container"
          >
            <svg
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 6v2h2l2 12H0L2 8h2V6a6 6 0 1 1 12 0zm-2 0a4 4 0 1 0-8 0v2h8V6zM4 10v2h2v-2H4zm10 0v2h2v-2h-2z"
              />
            </svg>
            <div
              class="amount"
            >
              <p>
                4
              </p>
            </div>
          </div>
        </div>
      </nav>
      <section
        class="cart-container"
      >
        <header>
          <h2>
            Your Bag
          </h2>
        </header>
        <div>
          <article
            class="cart-item"
          >
            <img
              alt="mockTitle0"
              src="mockImg0"
            />
            <div>
              <h4>
                mockTitle0
              </h4>
              <h4
                class="item-price"
              >
                100
              </h4>
              <button
                aria-label="remove"
                class="remove-btn"
              >
                remove
              </button>
            </div>
            <div>
              <button
                aria-label="increase"
                class="amount-btn"
              >
                <svg
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.707 7.05L10 6.343 4.343 12l1.414 1.414L10 9.172l4.243 4.242L15.657 12z"
                  />
                </svg>
              </button>
              <p
                class="amount"
              >
                2
              </p>
              <button
                aria-label="decrease"
                class="amount-btn"
              >
                <svg
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
                  />
                </svg>
              </button>
            </div>
          </article>
          <article
            class="cart-item"
          >
            <img
              alt="mockTitle1"
              src="mockImg1"
            />
            <div>
              <h4>
                mockTitle1
              </h4>
              <h4
                class="item-price"
              >
                170
              </h4>
              <button
                aria-label="remove"
                class="remove-btn"
              >
                remove
              </button>
            </div>
            <div>
              <button
                aria-label="increase"
                class="amount-btn"
              >
                <svg
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.707 7.05L10 6.343 4.343 12l1.414 1.414L10 9.172l4.243 4.242L15.657 12z"
                  />
                </svg>
              </button>
              <p
                class="amount"
              >
                1
              </p>
              <button
                aria-label="decrease"
                class="amount-btn"
              >
                <svg
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
                  />
                </svg>
              </button>
            </div>
          </article>
          <article
            class="cart-item"
          >
            <img
              alt="mockTitle2"
              src="mockImg2"
            />
            <div>
              <h4>
                mockTitle2
              </h4>
              <h4
                class="item-price"
              >
                150
              </h4>
              <button
                aria-label="remove"
                class="remove-btn"
              >
                remove
              </button>
            </div>
            <div>
              <button
                aria-label="increase"
                class="amount-btn"
              >
                <svg
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.707 7.05L10 6.343 4.343 12l1.414 1.414L10 9.172l4.243 4.242L15.657 12z"
                  />
                </svg>
              </button>
              <p
                class="amount"
              >
                1
              </p>
              <button
                aria-label="decrease"
                class="amount-btn"
              >
                <svg
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
                  />
                </svg>
              </button>
            </div>
          </article>
        </div>
        <footer>
          <hr />
          <div
            class="cart-total"
          >
            <h4>
              total 
              <span>
                $
                520.00
              </span>
            </h4>
          </div>
          <button>
            clear cart
          </button>
        </footer>
      </section>
    </main>
  </div>
</body>
`;

exports[`Cart Should render correctly loading 1`] = `
<body>
  <div>
    <main
      class="cart_project"
    >
      <nav>
        <div
          class="container"
        >
          <h3>
            Cart Project
          </h3>
          <div
            class="amount-container"
          >
            <svg
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 6v2h2l2 12H0L2 8h2V6a6 6 0 1 1 12 0zm-2 0a4 4 0 1 0-8 0v2h8V6zM4 10v2h2v-2H4zm10 0v2h2v-2h-2z"
              />
            </svg>
            <div
              class="amount"
            >
              <p>
                0
              </p>
            </div>
          </div>
        </div>
      </nav>
      <section
        class="cart-container"
      >
        <header>
          <h2>
            Your Bag
          </h2>
          <h4
            class="empty-cart"
          >
            is currently empty
          </h4>
        </header>
      </section>
    </main>
  </div>
</body>
`;
